keras/applications/convnext.py

```diff
@@ -0,0 +1,661 @@
+      if no `weights` argument is specified. Defaults to 1000 (number of
+      ImageNet classes).
+    classifier_activation: A `str` or callable. The activation function to use
+      on the "top" layer. Ignored unless `include_top=True`. Set
+      `classifier_activation=None` to return the logits of the "top" layer.
+      Defaults to `"softmax"`.
+      When loading pretrained weights, `classifier_activation` can only
+      be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+class StochasticDepth(layers.Layer):
+  """Stochastic Depth module. It performs batch-wise dropping rather than
+    sample-wise. In libraries like `timm`, it's similar to `DropPath` layers
+    that drops residual paths sample-wise.
+
+  Reference:
+    - https://github.com.rwightman/pytorch-image-models
+
+  Args:
+    drop_path_rate (float): Probability of dropping paths. Should be within
+      [0, 1].
+  
+  Returns:
+    Tensor either with the residual path dropped or kept.
+
+  """
+  def __init__(self, drop_path_rate, **kwargs):
+    super().__init__(**kwargs)
+    self.drop_path_rate = drop_path_rate
+
+  def call(self, x, training=False):
+    if training:
+      keep_prob = 1 - self.drop_path_rate
+      shape = (tf.shape(x)[0],) + (1,) * (len(tf.shape(x)) - 1)
+      random_tensor = keep_prob + tf.random.uniform(shape, 0, 1)
+      random_tensor = tf.floor(random_tensor)
+      return (x / keep_prob) * random_tensor
+    return x
+
+  def get_config(self):
+    config = super().get_config()
+    config.update({"drop_path_rate": self.drop_path_rate})
+    return config
+
+
+class ConvNeXtBlock(Model):
+  """ConvNeXt block.
+  
+  References:
+    - https://arxiv.org/abs/2201.03545
+    - https://github.com/facebookresearch/ConvNeXt/blob/main/models/convnext.py
+
+  Notes:
+    In the original ConvNeXt implementation (linked above), the authors use
+    `Dense` layers for pointwise convolutions for increased efficiency. 
+    Following that, this implementation also uses the same.
+
+  Args:
+    projection_dim (int): Number of filters for convolution layers. In the
+    ConvNeXt paper, this is referred to as projection dimension.
+    drop_path (float): Probability of dropping paths. Should be within [0, 1].
+    layer_scale_init_value (float): Layer scale value. Should be a small float
+      number.
+
+  Returns:
+    A keras.Model instance.
+  """
+  def __init__(self, projection_dim, drop_path_rate=0.0, 
+    layer_scale_init_value=1e-6, **kwargs):
+    super().__init__(**kwargs)
+    self.projection_dim = projection_dim
+    self.drop_path_rate = drop_path_rate
+    self.layer_scale_init_value = layer_scale_init_value
+    name = kwargs["name"]
+    
+    if layer_scale_init_value > 0.0:
+      self.gamma = tf.Variable(
+        layer_scale_init_value * tf.ones((projection_dim,)),
+        name=name + "_layer_scale_gamma")
+    else:
+      self.gamma = None
+    
+    self.dw_conv_1 = layers.Conv2D(
```

Comment:
It would be more in line with the keras style guide to remove these abbreviations:

dw_conv_1 ->
depthwise_conv_1